centreon/www/class/centreonConfigCentreonBroker.php

```diff
@@ -1340,24 +1340,30 @@ public function updateCentreonBrokerInfosByAPI(int $configId, array $values): vo
     }
 
     /**
-     * Generate fieldtype array.
+     * Generate fieldinfos array.
      *
      * @param int $typeId The type id
-     * @return array
+     * @return array<string,array<string,{type:string,default:string|null}>|{type:string,default:string|null}>
      */
-    public function getFieldtypesWithGroup($typeId)
+    public function getFieldInfosWithGroup($typeId)
     {
-        $fieldTypes = [];
+        $fields = [];
         $block = $this->getBlockInfos($typeId);
         foreach ($block as $fieldInfos) {
-            if ($fieldInfos['group_name'] !== null) {
-                $fieldTypes[$fieldInfos['group_name']][$fieldInfos['fieldname']] = $fieldInfos['fieldtype'];
+            if (! empty($fieldInfos['group_name']) && $fieldInfos['group_name'] !== null) {
```

Comment:
```suggestion
            if (! empty($fieldInfos['group_name'])) {
```
not needed as empty whill check null value